KVA/Migration.Tool.Source/Mappers/ContentItemMapper.cs

```diff
@@ -757,7 +785,30 @@ private IEnumerable<IUmtModel> MigrateDraft(ICmsVersionHistory checkoutVersion,
                 }
             }
 
-            // supply legacy document metadata fields
+            // Map legacy metadata fields
+            // Fields from custom class mapping
+            if (mapping is not null)
+            {
+                foreach (var fieldMapping in mapping.Mappings)
+                {
+                    var sourceField =
+                        legacyMetadataFields.Value.FirstOrDefault(x => x.LegacyFieldName == fieldMapping.SourceFieldName);
+                    if (sourceField is not null)

```

Comment:
If we remove the negation from the condition, we can avoid the arrowhead antipattern here. The if conditions across the whole solution are getting out of the control. They have multiple nested levels and it is really hard to read what is actually going on in the given method. Let's simplify the implementation for the future generations.